### PR TITLE
[INLONG-5533][Agent] Support structured output in the Kubernetes

### LINKED ISSUE #5533: [Improve][Agent] Support structured output in the Kubernetes
### Description

_No response_

### InLong Component

InLong Agent

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by GanfengTan**
File collects in the k8s, data, and metadata of k8s need to connect a message.

- Fixes #5533 

### Motivation
File data and k8s metadata to connect a message.

```
{
"_content_":"data",
"_time_":"time",
"_pod_name_":"test"
}
```

### Verifying this change

*(Please pick either of the following options)*

- [x] This change is a trivial rework/code cleanup without any test coverage.

### CODE REVIEW

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/AbstractFileReader.java [hunk 1]
```diff
@@ -35,9 +44,26 @@ public void mergeData(FileReaderOperator fileReaderOperator) {
         if (null == fileReaderOperator.metadata) {
             return;
         }
-
         List<String> lines = fileReaderOperator.stream.collect(Collectors.toList());
-        lines.forEach(data -> data = MetaDataUtils.concatString(data, fileReaderOperator.metadata));
+        if (fileReaderOperator.jobConf.hasKey(JOB_FILE_CONTENT_COLLECT_TYPE)) {
+            long timestamp = System.currentTimeMillis();
+            Gson gson = new Gson();
```
Comment (healchow) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/AbstractFileReader.java:50]: It is recommended to use a global static Gson object.

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/KubernetesFileReader.java [hunk 2]
```diff
@@ -60,22 +71,30 @@ public void getData() {
...
             return;
         }
-        client = getKubernetesClient();
-        Map<String, String> k8sInfo = MetaDataUtils.getLogInfo(fileReaderOperator.file.getName());
-        ObjectMeta objectMeta = getPodMetadata(k8sInfo.get(NAMESPACE), k8sInfo.get(POD_NAME));
-        fileReaderOperator.metadata = Objects.nonNull(objectMeta) ? objectMeta.toString() : null;
+        try {
+            client = getKubernetesClient();
+        } catch (IOException e) {
+            log.error("Get k8s client error: {}", e.getMessage());
```
Comment (healchow) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/KubernetesFileReader.java:77]: It is recommended to print all stack traces in error logs.

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/KubernetesFileReader.java [hunk 3]
```diff
@@ -92,11 +111,36 @@ public PodList getPods() {
...
+        if (Objects.isNull(namespaces) || namespaces.isEmpty()) {
+            return null;
+        }
+        if (!namespaces.contains(k8sInfo.get(NAMESPACE))) {
+            return null;
+        }
+        Pod pod = client.pods().inNamespace(k8sInfo.get(NAMESPACE)).withName(k8sInfo.get(POD_NAME)).get();
+        PodList podList = client.pods().inNamespace(k8sInfo.get(NAMESPACE))
+                .withLabels(MetaDataUtils.getPodLabels(jobConf)).list();
+        Gson gson = new Gson();
```
Comment (healchow) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/KubernetesFileReader.java:132]: It is recommended to use a global static Gson object.

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/FileDataUtils.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+/**
+ * File job utils
+ */
+public class FileDataUtils {
+
+    public static final  String KUBERNETES_LOG = "log";
+
+    // Get standard log for k8s
+    public static String getK8sJsonLog(String log) {
```
Comment (healchow) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/FileDataUtils.java:36]: For public static method, it is recommended to use Java doc.

Comment (healchow) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/FileDataUtils.java:36]: For public static method, it is recommended to use Java doc.

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/FileDataUtils.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        if (!StringUtils.isNoneBlank(log)) {
+            return null;
+        }
+        Gson gson = new Gson();
```
Comment (healchow) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/FileDataUtils.java:40]: Use a global static Gson object.

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/MetaDataUtils.java [hunk 6]
```diff
@@ -53,10 +62,59 @@ public static Map<String, String> getLogInfo(String fileName) {
...
+    }
+
+    public static List<String> getNamespace(JobProfile jobProfile) {
+        if (Objects.isNull(jobProfile) || !jobProfile.hasKey(JOB_FILE_PROPERTIES)) {
+            return null;
+        }
+        String property = jobProfile.get(JOB_FILE_PROPERTIES);
+        Type type = new TypeToken<HashMap<Integer, String>>() {
+        }.getType();
+        Gson gson = new Gson();
```
Comment (healchow) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/MetaDataUtils.java:88]: Use a global static Gson object.

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/MetaDataUtils.java [hunk 7]
```diff
@@ -53,10 +62,59 @@ public static Map<String, String> getLogInfo(String fileName) {
...
+     */
+    public static String getPodName(JobProfile jobProfile) {
+        if (Objects.isNull(jobProfile) || !jobProfile.hasKey(JOB_FILE_PROPERTIES)) {
+            return null;
         }
-        return str1.concat("\n").concat(str2);
+        String property = jobProfile.get(JOB_FILE_PROPERTIES);
+        Type type = new TypeToken<HashMap<Integer, String>>() {
+        }.getType();
+        Gson gson = new Gson();
```
Comment (healchow) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/MetaDataUtils.java:110]: Use a global static Gson object.

**Review by healchow — CHANGES_REQUESTED** [review]

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/AbstractFileReader.java (hunk 1 above)
Comment (GanfengTan) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/AbstractFileReader.java:50]: done

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/KubernetesFileReader.java (hunk 2 above)
Comment (GanfengTan) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/KubernetesFileReader.java:77]: done

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/KubernetesFileReader.java (hunk 3 above)
Comment (GanfengTan) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/KubernetesFileReader.java:132]: done

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/FileDataUtils.java (hunk 4 above)
Comment (GanfengTan) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/FileDataUtils.java:36]: done

Comment (GanfengTan) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/FileDataUtils.java:36]: done

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/FileDataUtils.java (hunk 5 above)
Comment (GanfengTan) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/FileDataUtils.java:40]: done

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/MetaDataUtils.java (hunk 6 above)
Comment (GanfengTan) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/MetaDataUtils.java:88]: done

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/MetaDataUtils.java (hunk 7 above)
Comment (GanfengTan) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/MetaDataUtils.java:110]: done

**Review by EMsnap — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Review by healchow — APPROVED** [review]

**Pull request merged**